### PR TITLE
Create event api

### PR DESCRIPTION
**Pull request by mustafaMari**

### CODE REVIEW

File: backend/src/main/java/planit/people/preparation/Google_Connector/Google_Helper.java [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
+import planit.people.preparation.Scheduling.Converter;
+import planit.people.preparation.Scheduling.Scheduler;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Calendar;
-import java.util.EventObject;
-import java.util.List;
+import java.util.*;
```
Comment (sereczeq) [inline backend/src/main/java/planit/people/preparation/Google_Connector/Google_Helper.java:17]: Don't import all. Speficy inports

File: backend/src/main/java/planit/people/preparation/Google_Connector/Google_Helper.java [hunk 2]
```diff
@@ -81,9 +81,59 @@ public CalendarResponse createEvent(DTO_NewEventDetail newEventDetail) throws IO
...
         event = google_connector.calendar_service().events().insert(calendarId, event).execute();
         System.out.printf("Event created: %s\n", event.getHtmlLink());
-        return new CalendarResponse(startDateTime, endDateTime);
+        return new CalendarResponse(startDate, endDateTime);
+    }
+
+    public DateTime getStartDate(Date start_date, Date end_date, Long duration) throws IOException
+    {
+        org.joda.time.DateTime sD = new org.joda.time.DateTime(start_date);
+        org.joda.time.DateTime eD = new org.joda.time.DateTime(end_date);
```
Comment (sereczeq) [inline backend/src/main/java/planit/people/preparation/Google_Connector/Google_Helper.java:90]: sD and start_date always use camelCase

File: backend/src/main/java/planit/people/preparation/Google_Connector/Google_Helper.java [hunk 3]
```diff
@@ -81,9 +81,59 @@ public CalendarResponse createEvent(DTO_NewEventDetail newEventDetail) throws IO
...
+        Duration durationInMinutes = Duration.standardMinutes(duration);
+        Interval first_interval = Scheduler.GetOneTimeSlotBetweenDatesOfLength(getFreeBusy(start_date, end_date), durationInMinutes, sD, eD);
+        Date event_start_date = first_interval.getStart().toDate();
```
Comment (sereczeq) [inline backend/src/main/java/planit/people/preparation/Google_Connector/Google_Helper.java:93]: same here, camelCase

File: backend/src/main/java/planit/people/preparation/Google_Connector/Google_Helper.java [hunk 4]
```diff
@@ -44,22 +45,21 @@ public CalendarResponse createEvent(DTO_NewEventDetail newEventDetail) throws IO
...
-        DateTime endDateTime = new DateTime(startDateTime.getValue() + (newEventDetail.duration() * 60 * 100));
+        DateTime endDateTime = new DateTime(startDate.getValue() + (newEventDetail.duration() * 60 * 1000));
         EventDateTime end = new EventDateTime()
                 .setDateTime(endDateTime)
-                .setTimeZone("Europe/Warsaw");
+                .setTimeZone(timeZone);
         event.setEnd(end);
 
-        String[] recurrence = new String[] {"RRULE:FREQ=DAILY;COUNT=1"};
+        String[] recurrence = new String[]{"RRULE:FREQ=DAILY;COUNT=1"};
```
Comment (sereczeq) [inline backend/src/main/java/planit/people/preparation/Google_Connector/Google_Helper.java:62]: is `"RRULE"` a spelling error or is it actually what's required.

Comment (sereczeq) [inline backend/src/main/java/planit/people/preparation/Google_Connector/Google_Helper.java:62]: Maybe declare that on top of file?

**Review by sereczeq — CHANGES_REQUESTED** [review]

File: backend/src/main/java/planit/people/preparation/Google_Connector/Google_Helper.java [hunk 5]
```diff
@@ -81,9 +81,59 @@ public CalendarResponse createEvent(DTO_NewEventDetail newEventDetail) throws IO
...
+        Date event_start_date = first_interval.getStart().toDate();
+        return new DateTime(event_start_date);
+    }
+
+    public Vector<Interval> getFreeBusy(Date startDate, Date endDate) throws IOException {
+        FreeBusyRequest req = new FreeBusyRequest();
+        req.setItems(getFreeBusyItems(getAllCalendarId()));
+        TimeZone tz = TimeZone.getTimeZone(timeZone);
+        DateTime startTime = new DateTime(startDate, tz);
+        DateTime endTime = new DateTime(endDate, tz);
```
Comment (Kaspr-whtev) [inline backend/src/main/java/planit/people/preparation/Google_Connector/Google_Helper.java:102]: could this be in a bit more consistent naming scheme? tz, startTime, endTime
I mean you even use 'timeZone' as a variable, why not stick to that?

**Review by Kaspr-whtev — CHANGES_REQUESTED** [review]

**Review by sereczeq — APPROVED** [review]
Great fix @sereczeq I'm proud of you

**Review by Kaspr-whtev — APPROVED** [review]

**Pull request merged**